java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -46,31 +40,25 @@ import arcs.core.storage.referencemode.toBridgingData
 import arcs.core.storage.referencemode.toBridgingOp
 import arcs.core.storage.referencemode.toBridgingOps
 import arcs.core.storage.referencemode.toReferenceModeMessage
+import arcs.core.storage.util.HoldQueue
+import arcs.core.storage.util.OperationQueue
 import arcs.core.storage.util.RandomProxyCallbackManager
-import arcs.core.storage.util.SendQueue
+import arcs.core.storage.util.SimpleQueue
 import arcs.core.type.Type
 import arcs.core.util.Random
 import arcs.core.util.Result
 import arcs.core.util.TaggedLog
 import arcs.core.util.computeNotNull
 import arcs.core.util.nextSafeRandomLong
-import kotlin.coroutines.CoroutineContext
-import kotlin.coroutines.coroutineContext
-import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CompletableDeferred
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.Job
 import kotlinx.coroutines.TimeoutCancellationException
-import kotlinx.coroutines.channels.ClosedSendChannelException
-import kotlinx.coroutines.channels.ConflatedBroadcastChannel
-import kotlinx.coroutines.flow.asFlow
-import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.filter
-import kotlinx.coroutines.flow.launchIn
-import kotlinx.coroutines.flow.onEach
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.withTimeout
 
+typealias BaseProxyMessage = ProxyMessage<CrdtData, CrdtOperationAtTime, Referencable>
+typealias RefModeProxyMessage = ProxyMessage<RefModeStoreData, RefModeStoreOp, RefModeStoreOutput>
```

Comment:
Some KDoc for what it means to be a BaseProxyMessage vs. a RefModeProxyMessage would be good on each of these (e.g. when they're used/why).